### ISSUE TITLE
Could not add downtime in service details (non-admin)

### ISSUE THREAD

**Issue by itsul** [centreon/centreon#5286]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Centreon Web version**: 2.8.8

**Centreon Engine version**: 1.7.2

**Centreon Broker version**: 3.0.7

**OS**: CES/CentOS 6.9

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical

**Steps to reproduce the issue:**
1.Login with non-admin user
2.Go to service details of random service
3.Add downtime for this service

**Describe the results you received:**
The Downtime is not written and the following error message appears in apache log:
`PHP Warning:  in_array() expects parameter 2 to be array, null given in /usr/share/centreon/www/include/monitoring/downtime/AddDowntime.php on line 386, referer: http://centhost/centreon/main.php?p=21001&o=a&host_id=47&service_id=199`

**Describe the results you expected:**
Downtime written and no PHP error.

**Additional information you think important (e.g. issue happens only occasionally):**
The problem is in: /usr/share/centreon/www/include/monitoring/downtime/AddDowntime.php

```
375             /*
376              * Set a downtime for a service list
377              */
378 
379             //catch fix input service_id
380             if (!is_array($_POST["service_id"])) {
381                 $_POST["service_id"] = array($_POST["service_id"]);
382             }
383
384             foreach ($_POST["service_id"] as $value) {
385                 $info = split('-', $value);
386                 if ($centreon->user->access->admin || in_array($info[0], $host_acl_id)) {
387                     $ecObj->addSvcDowntime(
388                         $info[0],
389                         $info[1],
390                         $_POST["comment"],
391                         $_POST["start"] . ' ' . $_POST['start_time'],
392                         $_POST["end"] . ' ' . $_POST['end_time'],
393                         $_POST["persistant"], 
394                         $duration,
395                         $host_or_centreon_time
396                     );  
397                 }   
398             }  
```
$host_acl_id in line 386 is not defined.
Adding 
`$host_acl_id = preg_split('/,/', str_replace("'", "", $hostStr));`
in line 383 over the loop fixes the issue. Hope I didn't miss any site effects.

**Comment by thiuyendang** [comment]
OK 2.8.9-11

**Comment by itsul** [comment]
Hi,
should these fixes be part of 2.8.9-11?
I think you forgot the milestone on this bug? (Sorry if I'm wrong,
still learning with GitHub :-) )
Cause we updated to 2.8.9-11 and the problem still exists.
Also both commits are still missing in the RPM.
I can confirm that it's working if we replace with the File from Git.

**Comment by Moujimouja** [comment]
Hello,

You're right, this has been implemented in a PR (https://github.com/centreon/centreon/pull/5289) but not merged yet on the 2.8.x branch.

We'll keep you posted

Have a nice afternoon
